### PR TITLE
All posts now being written to posts.json

### PR DESCRIPTION
**Pull request by willjw3**
No need to merge this PR. It's just so you can view the diff and see what changes were made. I changed the condition statement for deciding when to write the postlist array to a json file. Can't remember if I even thought of that when I went through the tutorial (it was largely off the cuff).

I can see that you might get an error in your `Featured.js` file. Let me know if you get stuck on that.

### CODE REVIEW

File: public/main.js [hunk 1]
```diff
@@ -44,6 +43,7 @@ const getPosts = () => {
           }
         };
 
+        const lines = contents.split("\n");
```
Comment (karamazov85) [inline public/main.js:46]: I wonder how this works - const variables are not hoisted, right? so how does parseMetadata and parseContent functions access this variable if lines were declared later?

File: public/main.js [hunk 2]
```diff
@@ -61,12 +61,14 @@ const getPosts = () => {
...
           content: content ? content : "No content given",
         };
-
+        
         postlist.push(post);
+        console.log(`length of postlist: ${postlist.length}`)
         // only if you are at the LAST iteration, write the postlist array into a JSON object
-        if (i === files.length - 1) {
+        // compare length of postlist array to length of files array 
+        if (postlist.length === files.length) {
```
Comment (karamazov85) [inline public/main.js:69]: Neat! I should have thought of this... what I don't understand is why the condition on line 67 doesn't work. It looks logical and clean and it clearly worked for you. Any hints as to why that doesn't work?

**Review by karamazov85 — COMMENTED** [review]
Thanks man, it's really nice of you to help out. 

I will merge this into master - but I am still finding my feet on Git so I'm looking at how to merge this properly. 

I added a few comments - questions that may help me understand why you did what you did.

Cheers!
Adam

File: public/main.js (hunk 1 above)
Comment (willjw3) [reply public/main.js:46]: What matters is when the functions are invoked. Of course, they need to be defined before invocation.

File: public/main.js (hunk 2 above)
Comment (willjw3) [reply public/main.js:69]: If you log `i` to the console, you'll see that due to asynchronous behavior, the values are not logged in order. To be honest, I don't remember what I was thinking when I made the file. I was planning on doing something different, but it seemed beyond the scope of the tutorial, so I changed things up. I got lucky, I guess.

**Comment by willjw3** [comment]
Like I mentioned before, I have a feeling you're going to run into a snag when you run your `Featured.js` file. Even if you don't, check the length of `post.featured`; you might be surprised.

**Comment by karamazov85** [comment]
Hey, 

Thanks man, you're awesome. Here is the link to the (more or less) finished website (which is an MVP, I will keep patching it up I guess):
http://www.pebblebeach-poetry.com/

If I wanted to change the approach and put all the poems up on, say, firebase, I would like to build an interface which would look like a proper text editor - one would type and format there, the text would be saved immediately to firebase as a document and then the website would pull them down from firebase real-time. 

I am planning to do this entirely with React but do you have any recommendations of other technologies that would make this easier? 

Many thanks again
Adam

**Comment by willjw3** [comment]
> If I wanted to change the approach and put all the poems up on, say, firebase, I would like to build an interface which would look like a proper text editor - one would type and format there, the text would be saved immediately to firebase as a document and then the website would pull them down from firebase real-time.
> 

I've never worked with Firebase before, so I'm not familiar with content management options for it. There's a CMS called Flamelink that people seem to recommend often. 
> I am planning to do this entirely with React but do you have any recommendations of other technologies that would make this easier?

Look into form handling with React; I'm sure you can find plenty of examples out there.

**Comment by willjw3** [comment]
I'm going to go ahead and close this PR. As I said, it was just so you could check out the diff. Hope all is working out the way you wanted it to.

**Pull request closed without merging**